packages/build/src/plugins/compatibility.js

```diff
@@ -52,6 +60,27 @@ const getCompatibleEntry = async function ({ versions, nodeVersion, packageJson,
   return compatibleEntry || { version: versions[0].version }
 }
 
+// Takes an array of versions and an object containing all the enabled feature
+// flags. It finds the first entry in versions that contains a `featureFlag`
+// field that matches one of the enabled feature flags — if one is found, that
+// version entry is bumped to the first slot, which effectively makes it the
+// new default version; if none is found, the default version is not changed.
+// Either way, any versions with a `featureFlag` field are filtered out before
+// returning, so that normal evaluation of conditions can take place downstream
+// without any concern for feature flags.
+const filterVersionsByFeatureFlag = ({ featureFlags = {}, versions }) => {
+  const match = versions.find(({ featureFlag }) => featureFlag !== undefined && featureFlags[featureFlag] === true)
+  const versionsWithoutFeatureFlags = versions.filter(({ featureFlag }) => featureFlag === undefined)
+
+  if (match === undefined) {
+    return versionsWithoutFeatureFlags
+  }
+
+  const newDefaultVersion = { ...match, featureFlag: undefined }
+
+  return [newDefaultVersion, ...versionsWithoutFeatureFlags]
```

Comment:
Do we want to bump any found feature flagged to the first slot?
For example, a plugin might have a more recent but non-feature flagged version, which should be preferred then.

Reply:
No, I think we should be explicit about all versions. Feature-flagged channels may be prerelease versions in any case, so we shouldn't use semver.